pilot/pkg/security/model/authentication.go

```diff
@@ -150,6 +154,13 @@ func ConstructSdsSecretConfig(name string) *tls.SdsSecretConfig {
 		return nil
 	}
 
+	var fetchTimeout *durationpb.Duration
+	if name == SDSDefaultResourceName || name == SDSRootResourceName {
+		fetchTimeout = ptypes.DurationProto(time.Second * 0)
+	} else {
+		fetchTimeout = features.InitialFetchTimeout
+	}
+
```

Comment:
I am wondering why you need to add this code logic here. As from the code if name == SDSDefaultResourceName or SDSRootResourceName the ConstructSdsSecretConfig function  will directly return the defaultSDSConfig and rootSDSConfig. So you only need to change the var field InitialFetchTimeout  of defaultSDSConfig and SDSRootResourceName to ptypes.DurationProto(time.Second * 0)

Reply:
@howardjohn indicated this was necessary, but I think you are right. Will change.

Reply:
oh, i see what happened. When I rebased this PR the logic to return rootSDSConfig or defaultSDSConfig were added, making my if statement redundant. OK, i removed it.